habitat-lab/habitat/datasets/rearrange/rearrange_generator.py

```diff
@@ -485,7 +510,22 @@ def generate_single_episode(self) -> Optional[RearrangeEpisode]:
         navmesh_path = osp.join(
             scene_base_dir, "navmeshes", scene_name + ".navmesh"
         )
-        self.sim.pathfinder.load_nav_mesh(navmesh_path)
+
+        # Load navmesh
+        if not self.cfg.regenerate_new_mesh:
+            self.sim.pathfinder.load_nav_mesh(navmesh_path)
```

Comment:
Thanks, Alex! Could we also regenerate the new mesh when `self.sim.pathfinder.load_nav_mesh(navmesh_path)` fails to load the navmesh (i.e., when it returns `False`)?

Reply:
Good idea.